osu.Game/Online/Chat/ChannelManager.cs

```diff
@@ -99,6 +102,28 @@ private void currentChannelChanged(ValueChangedEvent<Channel> e)
         /// </summary>
         private readonly Queue<Action> postQueue = new Queue<Action>();
 
+        /// <summary>
+        /// Scans for inline commands of the form $cmd and returns the newly processed text
+        /// </summary>
+        public string ProcessInlineCommands(string text) {
+
+            return Regex.Replace(text, @"\$[a-zA-Z0-9]+", delegate(Match match) {
+                string cmd = match.Value.Substring(1, match.Value.Length - 1);
+                switch (cmd) {
+                    case "np":
+                        BeatmapInfo beatmapInfo = currentBeatmap.Value.Beatmap.BeatmapInfo;
```

Comment:
this is incorrect

Reply:
what if i jumped into the chat overlay and typed `$np` before selecting a beatmap

Reply:
as in, `currentBeatmap.Value.BeatmapInfo` (you should not be accessing the `Beatmap` as this potentially causes a load/parse of the beatmap).

Reply:
how should I access the maps info without triggering a reload, I've been searching around and that's the only method I could find